extension/goplugin/schemas.go

```diff
@@ -123,8 +123,54 @@ func (schema *Schema) ID() string {
 	return schema.raw.ID
 }
 
-// StructToMap converts structure representation of the resource to mapped representation
-func (schema *Schema) StructToMap(resource interface{}) map[string]interface{} {
+// ResourceFromContext converts mapped representation to structure representation of the resource
+func (schema *Schema) ResourceFromContext(context map[string]interface{}) (res goext.Resource, err error) {
+	rawType, ok := schema.env.rawTypes[schema.ID()]
```

Comment:
I don't like the idea of having this as a member of Schema - it could complicate testing (think mocking). Why not make it a free function taking env as an argument? env already exposes RawTypes().